ipa-core/src/helpers/hashing.rs

```diff
@@ -74,9 +70,37 @@ where
         sha.update(&buf);
```

Comment:
Is there a risk that this is a bit churn-y.  Is there value in writing multiple items before updating the hash?  Or are we OK with our vectorization code.

Reply:
My guess is that this can optimize reasonably well. I'm inclined to leave it unless/until it shows up in profiling.